vowpalwabbit/cb_adf.cc

```diff
@@ -83,6 +88,110 @@ void learn_IPS(cb_adf& mydata, multi_learner& base, multi_ex& examples)
   call_cs_ldf<true>(base, examples, mydata.cb_labels, mydata.cs_labels, mydata.prepped_cs_labels, mydata.offset);
 }
 
+float safe_probability(float prob)
+{
+  if (prob <= 0.)
+  {
+    std::cout << "Probability " << prob << " is not possible, replacing with 1e-6.  Fix your dataset. " << std::endl;
```

Comment:
Error messages or warnings should be printed to cerr

Reply:
Made the following changes:
cb_adf.cc: Removed the safe_probability method. Replaced all calls to it to call GEN_CS::safe_probability instead. Having 2 safe_probability methods is not healthy. 
gen_cs_example.cc::safe_probability prints error message to std::cout and needs to be fixed in the master branch.